gprofiler/docker_client.py

```diff
@@ -22,43 +27,45 @@ def __init__(self):
             self._client = None
 
         self._pid_to_container_name_cache: Dict[int, str] = {}
-        self._container_id_to_name_cache: Dict[str, str] = {}
+        self._current_container_names: Set[str] = set()
 
     def reset_cache(self):
-        self._pid_to_container_name_cache = {}
-        self._container_id_to_name_cache = {}
+        self._pid_to_container_name_cache.clear()
+        self._current_container_names.clear()
 
-    def get_container_names(self) -> List[str]:
-        return list(self._container_id_to_name_cache.values())
+    @property
+    def container_names(self) -> List[str]:
+        return list(self._current_container_names)
 
-    def get_stack_line_with_container_name(self, stack_line: str, pid: int) -> str:
+    def get_container_name(self, pid: int) -> str:
         if self._client is None:
-            return f';{stack_line}'
+            return ''
         if pid in self._pid_to_container_name_cache:
-            cached_container_name = self._pid_to_container_name_cache[pid]
-            return f'{cached_container_name};{stack_line}'
-        container_name: Optional[str] = self._get_process_container_name(pid)
+            return self._pid_to_container_name_cache[pid]
+        container_name: Optional[str] = self._safely_get_process_container_name(pid)
         if container_name is None:
             self._pid_to_container_name_cache[pid] = ''
-            return f';{stack_line}'
+            return ''
         self._pid_to_container_name_cache[pid] = container_name
-        return f'{container_name};{stack_line}'
+        return container_name
 
-    def _get_process_container_name(self, pid: int) -> Optional[str]:
+    def _safely_get_process_container_name(self, pid: int) -> Optional[str]:
         try:
             container_id = self._get_process_container_id(pid)
-            if not container_id:
+            if container_id is None:
                 return None
-            if container_id in self._container_id_to_name_cache:
-                return self._container_id_to_name_cache[container_id]
-            container = self._client.containers.get(container_id)
-            container_name = container.name
-            self._container_id_to_name_cache[container_id] = container_name
-            return container_name
+            return self._get_container_name(container_id)
         except Exception:
             logger.warning(f'Could not get a container name for PID {pid}', exc_info=True)
             return None
 
+    @functools.lru_cache(4096)
+    def _get_container_name(self, container_id):
+        container = self._client.containers.get(container_id)
+        container_name = container.name
+        self._current_container_names.add(container_name)
```

Comment:
calling this function with the same `container_id` in a new run where the `lru_cache` is not empty might cause it to skip `self._current_container_names.add(container_name)` which should be called - move outside?

Reply:
Changed logic, so this is irrelevant